do 3 i = 1,10
    call foo(i)
3 continue